### ISSUE TITLE
Exception when calling sample_space with preset='synspace'

### ISSUE THREAD

**Issue by PatWalters** [ur-whitelab/exmol#129]
```
def rand_model(smi):
    return random.randint(0,1)

palbo_smi = "CC(=O)C1=C(C)c2cnc(Nc3ccc(cn3)N4CCNCC4)nc2N(C5CCCC5)C1=O"
exmol.sample_space(palbo_smi,rand_model,batched=False,preset='synspace')
```

--------------------------------------------------------------------------------------------

SMILESParserError                         Traceback (most recent call last)
File ~/miniconda3/envs/rdkit_2022_12/lib/python3.8/site-packages/selfies/encoder.py:65, in encoder(smiles, strict, attribute)
     64 try:
---> 65     mol = smiles_to_mol(smiles, attributable=attribute)
     66 except SMILESParserError as err:

File ~/miniconda3/envs/rdkit_2022_12/lib/python3.8/site-packages/selfies/utils/smiles_utils.py:214, in smiles_to_mol(smiles, attributable)
    213 mol = MolecularGraph(attributable=attributable)
--> 214 tokens = deque(tokenize_smiles(smiles))
    215 i = 0

File ~/miniconda3/envs/rdkit_2022_12/lib/python3.8/site-packages/selfies/utils/smiles_utils.py:119, in tokenize_smiles(smiles)
    118     err_msg = "unrecognized symbol '{}'".format(smiles[i])
--> 119     raise SMILESParserError(smiles, err_msg, i)
    121 yield token

SMILESParserError: 
	SMILES: CC(=O)c1c(C)c2cnc(Nc3ccc(N(C)Cc4n~c5nc(C)ccc5n4C)cn3)nc2n(C2CCCC2)c1=O
	                                         ^
	Index:  33
	Reason: unrecognized symbol '~'

The above exception was the direct cause of the following exception:

EncoderError                              Traceback (most recent call last)
Cell In[21], line 5
      2     return random.randint(0,1)
      4 palbo_smi = "CC(=O)C1=C(C)c2cnc(Nc3ccc(cn3)N4CCNCC4)nc2N(C5CCCC5)C1=O"
----> 5 exmol.sample_space(palbo_smi,rand_model,batched=False,preset='synspace')

File ~/miniconda3/envs/rdkit_2022_12/lib/python3.8/site-packages/exmol/exmol.py:687, in sample_space(origin_smiles, f, batched, preset, data, method_kwargs, num_samples, stoned_kwargs, quiet, use_selfies, sanitize_smiles)
    683     data = [mol2smi(mol) for mol in mols]
    684     smiles, scores = run_custom(
    685         origin_smiles, data=cast(Any, data), _pbar=pbar, **method_kwargs
    686     )
--> 687     selfies = [sf.encoder(s) for s in smiles]
    688 else:
    689     result = run_stoned(
    690         origin_smiles, _pbar=pbar, return_selfies=True, **method_kwargs
    691     )

File ~/miniconda3/envs/rdkit_2022_12/lib/python3.8/site-packages/exmol/exmol.py:687, in <listcomp>(.0)
    683     data = [mol2smi(mol) for mol in mols]
    684     smiles, scores = run_custom(
    685         origin_smiles, data=cast(Any, data), _pbar=pbar, **method_kwargs
    686     )
--> 687     selfies = [sf.encoder(s) for s in smiles]
    688 else:
    689     result = run_stoned(
    690         origin_smiles, _pbar=pbar, return_selfies=True, **method_kwargs
    691     )

File ~/miniconda3/envs/rdkit_2022_12/lib/python3.8/site-packages/selfies/encoder.py:68, in encoder(smiles, strict, attribute)
     66 except SMILESParserError as err:
     67     err_msg = "failed to parse input\n\tSMILES: {}".format(smiles)
---> 68     raise EncoderError(err_msg) from err
     70 if not mol.kekulize():
     71     err_msg = "kekulization failed\n\tSMILES: {}".format(smiles)

EncoderError: failed to parse input
	SMILES: CC(=O)c1c(C)c2cnc(Nc3ccc(N(C)Cc4n~c5nc(C)ccc5n4C)cn3)nc2n(C2CCCC2)c1=O

⚡CUSTOM⚡: : 1094it [00:23, 279.20it/s]

**Comment by PatWalters** [comment]
It looks like this is actually a synspace bug

```
palbo_smi = "CC(=O)C1=C(C)c2cnc(Nc3ccc(cn3)N4CCNCC4)nc2N(C5CCCC5)C1=O"
mols, props = synspace.chemical_space(palbo_smi)
smi_list = [Chem.MolToSmiles(x) for x in mols]
[x for x in smi_list if x.find("~") >= 0]

['CC(=O)c1c(C)c2cnc(Nc3ccc(N(C)Cc4n~c5cc(C)cnc5n4C)cn3)nc2n(C2CCCC2)c1=O',
 'CC(=O)c1c(C)c2cnc(Nc3ccc(N(C)Cc4n~c5ccc(C)nc5n4C)cn3)nc2n(C2CCCC2)c1=O',
 'CC(=O)c1c(C)c2cnc(Nc3ccc(N(C)Cc4n~c5ncccc5n4C)cn3)nc2n(C2CCCC2)c1=O',
 'CC(=O)c1c(C)c2cnc(Nc3ccc(N(C)Cc4n~c5cccnc5n4C)cn3)nc2n(C2CCCC2)c1=O',
 'CC(=O)c1c(C)c2cnc(Nc3ccc(N(C)Cc4n~c5cnccc5n4C)cn3)nc2n(C2CCCC2)c1=O',
 'CCn1c(CN(C)c2ccc(Nc3ncc4c(C)c(C(C)=O)c(=O)n(C5CCCC5)c4n3)nc2)n~c2cccnc21',
 'CC(=O)c1c(C)c2cnc(Nc3ccc(N(C)Cc4n~c5cccc6c5n4CCO6)cn3)nc2n(C2CCCC2)c1=O',
 'CC(=O)c1c(C)c2cnc(Nc3ccc(N(C)Cc4n~c5ccc(O)cc5n4C)cn3)nc2n(C2CCCC2)c1=O',
 'CC(=O)c1c(C)c2cnc(Nc3ccc(N(C)Cc4n~c5cccc(O)c5n4C)cn3)nc2n(C2CCCC2)c1=O',
 'COc1cccc2c1~nc(CN(C)c1ccc(Nc3ncc4c(C)c(C(C)=O)c(=O)n(C5CCCC5)c4n3)nc1)n2C',
 'COc1ccc2c(c1)~nc(CN(C)c1ccc(Nc3ncc4c(C)c(C(C)=O)c(=O)n(C5CCCC5)c4n3)nc1)n2C']
```

**Comment by PatWalters** [comment]
I think there's a problem with the reaction SMARTS for 

`benzimidazole_derivatives_aldehyde`
Why is there an "@" sign at the end of the SMARTS.  The product looks like this
`[c:3]2:[c:1]:[n:2]:[c:5](-[#6:6]):[n:4]@2`
I don't think the @ before the 2 at the end is wrong.

**Comment by PatWalters** [comment]
I noticed that 5 of the reaction SMARTS have "@" in the products.  Is this correct? 

```
 "benzimidazole_derivatives_carboxylic-acid/ester": "[c;r6:1](-[NH1;$(N-[#6]):2]):[c;r6:3](-[NH2:4]).[#6:6]-[C;R0:5](=[OD1])-[#8;H1,$(O-[CH3])]>>[c:3]2:[c:1]:[n:2]:[c:5](-[#6:6]):[n:4]@2",
 "benzimidazole_derivatives_aldehyde": "[c;r6:1](-[NH1;$(N-[#6]):2]):[c;r6:3](-[NH2:4]).[#6:6]-[CH1;R0:5](=[OD1])>>[c:3]2:[c:1]:[n:2]:[c:5](-[#6:6]):[n:4]@2",
 "benzothiazole": "[c;r6:1](-[SH1:2]):[c;r6:3](-[NH2:4]).[#6:6]-[CH1;R0:5](=[OD1])>>[c:3]2:[c:1]:[s:2]:[c:5](-[#6:6]):[n:4]@2",
 "benzoxazole_arom-aldehyde": "[c:1](-[OH1;$(Oc1ccccc1):2]):[c;r6:3](-[NH2:4]).[c:6]-[CH1;R0:5](=[OD1])>>[c:3]2:[c:1]:[o:2]:[c:5](-[c:6]):[n:4]@2",
 "benzoxazole_carboxylic-acid": "[c;r6:1](-[OH1:2]):[c;r6:3](-[NH2:4]).[#6:6]-[C;R0:5](=[OD1])-[OH1]>>[c:3]2:[c:1]:[o:2]:[c:5](-[#6:6]):[n:4]@2",
```

**Comment by whitead** [comment]
Fixed on exmol for now with b2bbcc916599865d8f3e5acea605cd0870025607. Will work more on the SMRIKS in the issue mentioned on synspace.
